File: Easy/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1 if x >= 0 else -1
        x = abs(x)
        
        rev = sign * int(str(x)[::-1])
        
        if rev < pow(-2, 31) or rev > pow(2, 31)-1:
            return 0
        else:
            return rev

File: Easy/Swift/reverse_integer.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        let minRange = -2_147_483_648
        let maxRange = 2_147_483_647

        let sign = x>0 ? 1 : -1
        let abs_x = String(abs(x)).reversed()

        let rev = sign * Int(String(abs_x))!

        if rev < minRange || rev > maxRange {
            return 0
        } else {
            return rev
        }
    }
}
